core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2956,7 +2956,8 @@ class ViewMapping< DstTraits , SrcTraits ,
     };
 
 public:
-
+  enum { is_assignable_data_type = is_assignable_value_type &&
```

Comment:
@crtrott I'm assuming this change will break Sacado, I'll update on the kokkos-promotion branch.